compiler/rustc_error_codes/src/error_codes/E0376.md

```diff
@@ -1,4 +1,7 @@
-`CoerceUnsized` was implemented on something that isn't a struct.
+#### Note: this error code is no longer emitted by the compiler.
```

Comment:
E0376 was folded into E0377, since it's a bit strange to split out the error that would be issued for `impl<T, U> CoerceUnsized<U> for W<T>` and `impl<T, U> CoerceUnsized<X<U>> for W<T>`. The root cause of both is that they need to be of the same ADT.